src/Components/Blazor/Server/src/Services/BlazorHostingServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net.Mime;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.StaticFiles;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class BlazorHostingServiceCollectionExtensions
+    {
+        public static IServiceCollection AddBlazorStaticFilesConfiguration(this IServiceCollection services)
+        {
+            services.Configure<StaticFileOptions>(options =>
+            {
+            });
+
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IConfigureOptions<StaticFileOptions>, ClientSideBlazorStaticFilesConfiguration>());
+
+            return services;
+
+        }
+
+        private class ClientSideBlazorStaticFilesConfiguration : IConfigureOptions<StaticFileOptions>
+        {
+            public ClientSideBlazorStaticFilesConfiguration(IWebHostEnvironment webHostEnvironment)
+            {
+                WebHostEnvironment = webHostEnvironment;
+            }
+
+            public IWebHostEnvironment WebHostEnvironment { get; }
+
+            public void Configure(StaticFileOptions options)
+            {
+                options.FileProvider = WebHostEnvironment.WebRootFileProvider;
+                var contentTypeProvider = new FileExtensionContentTypeProvider();
+                AddMapping(contentTypeProvider, ".dll", MediaTypeNames.Application.Octet);
+                // For right now unconditionally enable debugging
```

Comment:
We will change this to be based on IConfiguration and  the enable debugging setting `https://github.com/dotnet/aspnetcore/pull/18368/files#diff-31f2ddf11f40d366c6e0d7d39f3cf06fR2`

Reply:
Follow up issue please.

Reply:
Updated this to use the setting from config.

Reply:
Not sure which config source you mean, and not sure there is one that gives the info you need here. Let's discuss this.

Reply:
Discussed offline, we can map this unconditionally as the pdbs will not be part of the build output unless you explicitly ask to.